modules/rollup-plugin-workbox-inject/index.js

```diff
@@ -31,39 +33,74 @@ const chalk = require('chalk');
+            file: upath.resolve(outputDir, chunk.fileName),
             baseDirectory: options.globDirectory,
-          }),
-        );
+          });
+          // Sourcemaps might be disabled, but adding something extra to
+          // globIgnores doesn't hurt.
+          const mapFile = `${fileRebasedToGlobDirectory}.map`;
+
+          options.globIgnores.push(
+            fileRebasedToGlobDirectory,
+            mapFile,
+          );
+
+          // If set, this should be the path to the original source module.
+          if (chunk.facadeModuleId) {
+            const moduleRebasedToGlobDirectory = rebasePath({
+              file: chunk.facadeModuleId,
+              baseDirectory: options.globDirectory,
+            });
+            options.globIgnores.push(moduleRebasedToGlobDirectory);
+          }
+        }
       }
-      const globalRegexp = new RegExp(escapeRegexp(options.injectionPoint), 'g');
 
       const { manifestEntries, count, warnings, size } = await getFileManifestEntries(options);
 
       const manifestString = stringify(manifestEntries);
 
-      const bundles = Object.keys(bundle);
+      for (const chunk of Object.values(bundle)) {
+        // This might actually be an asset (which won't have a code property)
+        // and not a chunk. If so, continue.
+        if (!chunk.code) {
+          continue;
+        }
+
+        if (chunk.code.includes(injectionPoint)) {
+          if (chunk.map) {
+            const { map, source } = await replaceAndUpdateSourceMap({
+              originalMap: chunk.map,
+              jsFilename: chunk.fileName,
+              originalSource: chunk.code,
+              replaceString: manifestString,
+              searchString: injectionPoint,
+            });
+            chunk.code = source;
+            chunk.map = map;
+          } else {
+            chunk.code = chunk.code.replace(injectionPoint, manifestString);
+          }
 
-      while (bundles.length) {
-        const bndl = bundle[bundles.pop()];
-        if (bndl.code.match(globalRegexp)) {
-          const { map, source } = await replaceAndUpdateSourceMap({
-            originalMap: bndl.map,
-            jsFilename: bndl.fileName,
-            originalSource: bndl.code,
-            replaceString: manifestString,
-            searchString: options.injectionPoint,
-          });
-          bndl.code = source;
-          bndl.map = map;
           const warn = `\nThere ${warnings.length > 1 ? 'were' : 'was'} ${chalk.red(warnings.length)} ${warnings.length > 1 ? 'warnings' : 'warning'}.`;
-          console.log(`Injected ${chalk.cyan(count + ' files')} for precaching, ${chalk.cyan(size * 0.001 + ' KB')} total, into ${chalk.cyan(bndl.fileName)}.${warnings.length ? warn : ''}`);
+          console.log(`Injected ${chalk.cyan(count + ' files')} for precaching, ${chalk.cyan(size * 0.001 + ' KB')} total, into ${chalk.cyan(chunk.fileName)}.${warnings.length ? warn : ''}`);
```

Comment:
Can I ask you do me a favor here while you're in the codebase? Can you round the output size to the nearest `.00`? I'm seeing lots of weird JS number output and that was next on my list to fix.

Reply:
Mind if I just add in https://www.npmjs.com/package/pretty-bytes?

That's what we use in `workbox-cli` and it's quite readable.

Reply:
Output looks like:

`Injected 24 files for precaching, 99.2 kB total, into service-worker.js.`